src/Common/src/System/Runtime/InteropServices/StringBuffer.cs

```diff
@@ -42,6 +42,20 @@ public unsafe StringBuffer(string initialContents)
         }
 
         /// <summary>
+        /// Instantiate the buffer with a copy of the specified string.
+        /// </summary>
+        public unsafe StringBuffer(StringBuffer initialContents)
+            : base(0)
+        {
+            // We don't pass the count of bytes to the base constructor, appending will
+            // initialize to the correct size for the specified initial contents.
+            if (initialContents != null)
+            {
+                Append(initialContents);
+            }
```

Comment:
Yes, we don't technically have to have it. Just simpler usage is all- and matches the constructor taking a string- which we don't technically need either.